### ISSUE TITLE
no available topology found

### ISSUE THREAD

**Issue by zerowebcorp** [openebs/cstor-operators#463]
Getting this error with a fresh installation. 

`failed to provision volume with StorageClass "cstor-csi-disk": error generating accessibility requirements: no available topology found`

Installation was using helm chart.

`values.yaml`
```
cstor:
  enabled: true
  csiNode:
    kubeletDir: /var/snap/microk8s/common/var/lib/kubelet
```

```
helm upgrade --install openebs --namespace openebs openebs/openebs \
--version 3.10.0 -f openebs.yaml
```

`diskpool.yaml`

```
apiVersion: cstor.openebs.io/v1
kind: CStorPoolCluster
metadata:
 name: cstor-disk-pool
 namespace: openebs
spec:
 pools:
   - nodeSelector:
       kubernetes.io/hostname: "ueazrcephnp01"
     dataRaidGroups:
       - blockDevices:
           - blockDeviceName: "blockdevice-92d13577847843b0fc3c732eb2d6f5a3"
     poolConfig:
       dataRaidGroupType: "stripe"

   - nodeSelector:
       kubernetes.io/hostname: "ueazrcephnp02"
     dataRaidGroups:
       - blockDevices:
           - blockDeviceName: "blockdevice-d42fff9a7ba34f4ace7ebf7ec056aa0d"
     poolConfig:
       dataRaidGroupType: "stripe"

   - nodeSelector:
       kubernetes.io/hostname: "ewazrcephnp01"
     dataRaidGroups:
       - blockDevices:
           - blockDeviceName: "blockdevice-317d380d3988ad6a14511a3412f0541b"
     poolConfig:
       dataRaidGroupType: "stripe"
   - nodeSelector:
       kubernetes.io/hostname: "ewazrcephnp02"
     dataRaidGroups:
       - blockDevices:
           - blockDeviceName: "blockdevice-93a9a8488451b5b8eac95c86102a1d64"
     poolConfig:
       dataRaidGroupType: "stripe"
```

`storageclass.yaml`

```
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  name: cstor-csi-disk
provisioner: cstor.csi.openebs.io
allowVolumeExpansion: true
parameters:
  cas-type: cstor
  # cstorPoolCluster should have the name of the CSPC
  cstorPoolCluster: cstor-disk-pool
  # replicaCount should be <= no. of CSPI created in the selected CSPC
  replicaCount: "3"
```

Upon creating a PVC, this is the error it throws

![image](https://github.com/openebs/cstor-operators/assets/914305/ead3bf2c-4235-4d50-9ca6-eb783862a623)

![image](https://github.com/openebs/cstor-operators/assets/914305/2a600a52-c2a6-4e6b-86c9-8f84590fcbcc)

![image](https://github.com/openebs/cstor-operators/assets/914305/165a8b7d-461b-4f2b-b1df-59716daf2c26)

Any clue on what could be the issue?

**Comment by tidux** [comment]
I'm seeing the same error.  Underlying cluster is K0s on ARM64 (Fedora 39).

**Comment by fabiorleal** [comment]
I don't know if it's still worth answering this issue, but I had the same problem when I started installing OpenEBS Cstor on my cluster.
I was also using K0s, but AMD64 and not ARM, and the problem is in the openebs-cstor-csi-node Daemonset (the name is default, if you installed it in the default way).
In the manifest of this daemonset, replace everything that is /var/lib/kubelet with /var/lib/k0s/kubelet, because K0s stores the kubelet inside its own folder.
Through Helm you can set this by passing --set csiNode.kubeletDir="/var/lib/k0s/kubelet/" or change it in values.yaml.

**Comment by tidux** [comment]
The OpenEBS docs include warning about the Kubelet directory and the argument fix so I think this issue can be resolved.

On May 20, 2025 10:32:39 AM PDT, fabiorleal ***@***.***> wrote:
>fabiorleal left a comment (openebs-archive/cstor-operators#463)
>
>I don't know if it's still worth answering this issue, but I had the same problem when I started installing OpenEBS Cstor on my cluster.
>I was also using K0s, but AMD64 and not ARM, and the problem is in the openebs-cstor-csi-node Daemonset (the name is default, if you installed it in the default way).
>In the manifest of this daemonset, replace everything that is /var/lib/kubelet with /var/lib/k0s/kubelet, because K0s stores the kubelet inside its own folder.
>Through Helm you can set this by passing --set csiNode.kubeletDir="/var/lib/k0s/kubelet/" or change it in values.yaml.
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/openebs-archive/cstor-operators/issues/463#issuecomment-2895286432
>You are receiving this because you commented.
>
>Message ID: ***@***.***>
-- 
Jonathan Lane
